### PR TITLE
Optimize.plugins

### PR DESCRIPTION
**Pull request by dsoffiantini**
Adds a suite of validation plugins for Optimize for Griffon

This code change adds a suite of validations to make sure optimize is configured correctly by the user, as well as events are flowing correctly for the plugin to function.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: plugins/aep-optimize-mbox-validation/index.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*
```
Comment (amadt) [inline plugins/aep-optimize-mbox-validation/index.ts:1]: Feels like the name of this file doesn't match what it actually does

File: plugins/aep-optimize-prerequisites/index.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (amadt) [inline plugins/aep-optimize-prerequisites/index.ts:1]: Is this one a duplicate of  `plugins/aep-optimize-peer-dependencies/index.ts`?

Comment (dsoffiantini) [reply plugins/aep-optimize-prerequisites/index.ts:1]: @amadt  Peer Dependencies makes sure you have offers or target installed, this one makes sure you have edge and optimize set up. I ended up splitting them up to give better feedback in the dashboard, but I have no issue combining them back together if you think that'd be good.

Comment (amadt) [reply plugins/aep-optimize-prerequisites/index.ts:1]: Nope. That seems fine

File: plugins/aep-optimize-mbox-validation/index.ts (hunk 1 above)
Comment (dsoffiantini) [reply plugins/aep-optimize-mbox-validation/index.ts:1]: @amadt Hmm basically this just makes sure that each requested mbox appears in the response. Would something like `mbox-response-events` or `mbox-response-validation` work better?

Comment (amadt) [reply plugins/aep-optimize-mbox-validation/index.ts:1]: I just didn't see anything specific to an mbox. It looks for decision scopes, but those could be an mbox or an offer. Maybe that's where I was confused.

Guess I'm not totally sure the use case. Maybe we're validation that if the user requests a certain mbox or offer, that something comes back in the response? Like if a person requests an mbox of TGT, but that mbox doesn't exist?

Comment (dsoffiantini) [reply plugins/aep-optimize-mbox-validation/index.ts:1]: @amadt Ah that makes sense, I just looked at the payloads swarna provided and thought the scopes were only associated with mboxes. Would there be a better way to validate that each requested MBox has a result? That was the task I was trying to address.

**Review by amadt — APPROVED** [review]

**Pull request merged**